bridge/PendingJobsAsyncWaitTask.h

```diff
@@ -0,0 +1,43 @@
+
+#pragma once
+
+#include "jobmanager/AsyncWaiter.h"
+#include "bridge/engine/UntypedEngineObject.h"
+#include "logger/Logger.h"
+#include <memory>
+#include <functional>
+#include <string>
+
+namespace transport
+{
+class Transport;
+}
+
+namespace bridge
+{
+
+class PendingJobsAsyncWaitTask : public jobmanager::AsyncWaitTask
+{
+public:
+    using OnTaskEndHandler = std::function<void()>;
+
+    PendingJobsAsyncWaitTask(const logger::LoggableId& loggableId,
+        const std::shared_ptr<const transport::Transport>& _transport,
+        const std::shared_ptr<UntypedEngineObject>& streamToHold,
```

Comment:
Was it imposible to make this a template to avoid poluting the objects we want to store here. It inherits AsyncWaitTask so it still fits into the AsyncWaiter.
You usually pass lambda into task handler, and we do mainly use it for getting destructor called when all references are gone. Am I missing something?